app/models/user.rb

```diff
@@ -1,5 +1,6 @@
 class User < ActiveRecord::Base
   authenticates_with_sorcery!
+  before_create :create_organiser_token
```

Comment:
just add `hidden_field_tag` for the role when adding new email - that way you'll create user with role organizer so you can make such check ;)